### PR TITLE
Enhance model preparer support

### LINKED ISSUE #1998: Enhance model preparer support
Adding the following support:

1. Support for following ops: floordivide, chunk, norm, avgpool2d, batchnorm, and groupnorm in model preparer.
2. Add transpose and reshape_as as math invariant ops
3. Modules with args and kwargs are now initialized in elementwise_ops using a common custom metaclass.
4. Added test cases for chunk and batch_norm in test_model_preparer.

### PR DESCRIPTION
**Pull request by quic-aishgupt**
This change adds the following:

1. Support for following ops: floordivide, chunk, norm, avgpool2d, batchnorm, and groupnorm in model preparer.
2. Add transpose and reshape_as as math invariant ops
3. Modules with args and kwargs are now initialized in elementwise_ops using a common custom metaclass.
4. Added test cases for chunk and batch_norm in test_model_preparer.

### CODE REVIEW

**Review by quic-akhobare — APPROVED** [review]

**Comment by quic-akhobare** [comment]
Hi @quic-aishgupt - in the future, please squash commits to meaningful commits. In this PR there are 4 commits, with the same message. This is not a good idea as you can tell.
